### ISSUE TITLE
Incorrectly marks variables updated in a while loop

### ISSUE THREAD

**Issue by kageurufu** [afonasev/flake8-return#54]
* Date you used flake8-return: Jul 6, 2020
* flake8-return version used, if any: 1.1.1
* Python version, if any: Python 3.8.3
* Operating System: Arch Linux on Linux 5.7.4

### Description

flake8-return seemingly fails to detect changes to a variable modified in a while loop.

### What I Did

```python
def doit():
    test = 1
    i = 4

    while i := i - 1:
        test = test + 1

    return test

class Node:
    def __init__(self, v, parent=None):
        self.v = v
        self.parent = parent
        self.child = Node(v - 1, parent=self) if v else None

    def __str__(self):
        name = f"Node<{self.v}>"
        node = self

        while node := node.parent:
            name = f"Node<{node.v}>.{name}"

        return name

    def tail(self):
        node = self
        while node.child:
            node = node.child
        return node

print(Node(5).tail())
```

A R504 error is detected at each return there, in `doit()`, `Node.__str__`, and `Node.tail`
